File: interviewbit-trees-invert-the-binary-tree.py
# Definition for a  binary tree node
# class TreeNode:
#    def __init__(self, x):
#        self.val = x
#        self.left = None
#        self.right = None

class Solution:
    # @param A : root node of tree
    # @return the root node in the tree
    def recursive_invert(self, node):
        if node:
            node.left, node.right = node.right, node.left
            self.invertTree(node.left)
            self.invertTree(node.right)

    def invertTree(self, A):
        self.recursive_invert(A)
        return A

File: interviewbit-trees-invert-the-binary-tree.java
/**
 * Definition for binary tree
 * class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) {
 *      val = x;
 *      left=null;
 *      right=null;
 *     }
 * }
 */
public class Solution {
    public void invert(TreeNode node) {
        if (node != null) {
            TreeNode temp = node.left;
            node.left = node.right;
            node.right = temp;
            invert(node.left);
            invert(node.right);
        }
    }
    public TreeNode invertTree(TreeNode A) {
        invert(A);
        return A;
    }
}
